packages/next/src/build/utils.ts

```diff
@@ -1360,7 +1360,7 @@ export async function buildAppStaticPaths({
   return StaticGenerationAsyncStorageWrapper.wrap(
     ComponentMod.staticGenerationAsyncStorage,
     {
-      urlPathname: page,
+      url: { pathname: page },
```

Comment:
Feel we should use another variable instead of `pathname` here since for url the pathname is always part of url. `page` sounds like it could be the path before normalized such as `/blog/[dynamic]`.